### PR TITLE
Implemented rrspacing

### PR DESCRIPTION
**Pull request by parth121101**
fixes: #3622

### CODE REVIEW

File: integration_tests/intrinsics_178.f90 [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    real :: z
+    real(dp) :: w
+    real(dp) :: x 
+
+    x = 4.1
+    y = 7.8
+    z = 3.2
+    w = 5.6
+
+    print *, Rrspacing(4.1)
```
Comment (certik) [inline integration_tests/intrinsics_178.f90:13]: Use lowercase everywhere:
```suggestion
    print *, rrspacing(4.1)
```

**Review by certik — APPROVED** [review]
Otherwise ok.

**Comment by certik** [comment]
I am going to merge it, you can send a new PR with the lowercase change.

**Comment by gxyd** [comment]
This generates a few warnings while building now after this PR:
```bash
In file included from /Users/gxyd/OpenSource/lfortran/src/lfortran/semantics/ast_symboltable_visitor.cpp:16:
In file included from /Users/gxyd/OpenSource/lfortran/src/lfortran/semantics/ast_common_visitor.h:10:
In file included from /Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_function_registry.h:4:
In file included from /Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_function_registry_util.h:4:
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3416:18: warning: variable 'fraction' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
        else if (kind == 8) {
                 ^~~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3430:29: note: uninitialized use occurs here
        fraction = std::abs(fraction);
                            ^~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3416:14: note: remove the 'if' if its condition is always true
        else if (kind == 8) {
             ^~~~~~~~~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3400:24: note: initialize the variable 'fraction' to silence this warning
        double fraction;
                       ^
                        = 0.0
1 warning generated.
[ 63%] Building CXX object src/lfortran/CMakeFiles/lfortran_lib.dir/semantics/ast_body_visitor.cpp.o
In file included from /Users/gxyd/OpenSource/lfortran/src/lfortran/semantics/ast_body_visitor.cpp:16:
In file included from /Users/gxyd/OpenSource/lfortran/src/lfortran/semantics/ast_common_visitor.h:10:
In file included from /Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_function_registry.h:4:
In file included from /Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_function_registry_util.h:4:
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3416:18: warning: variable 'fraction' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
        else if (kind == 8) {
                 ^~~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3430:29: note: uninitialized use occurs here
        fraction = std::abs(fraction);
                            ^~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3416:14: note: remove the 'if' if its condition is always true
        else if (kind == 8) {
             ^~~~~~~~~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3400:24: note: initialize the variable 'fraction' to silence this warning
        double fraction;
                       ^
                        = 0.0
1 warning generated.
[ 64%] Building CXX object src/lfortran/CMakeFiles/lfortran_lib.dir/semantics/ast_to_asr.cpp.o
In file included from /Users/gxyd/OpenSource/lfortran/src/lfortran/semantics/ast_to_asr.cpp:13:
In file included from /Users/gxyd/OpenSource/lfortran/src/lfortran/semantics/ast_common_visitor.h:10:
In file included from /Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_function_registry.h:4:
In file included from /Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_function_registry_util.h:4:
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3416:18: warning: variable 'fraction' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
        else if (kind == 8) {
                 ^~~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3430:29: note: uninitialized use occurs here
        fraction = std::abs(fraction);
                            ^~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3416:14: note: remove the 'if' if its condition is always true
        else if (kind == 8) {
             ^~~~~~~~~~~~~~~
/Users/gxyd/OpenSource/lfortran/src/libasr/pass/intrinsic_functions.h:3400:24: note: initialize the variable 'fraction' to silence this warning
        double fraction;
                       ^
                        = 0.0
```

don't seem too hard to get rid of them though.

**Pull request merged**